index.html

```diff
@@ -1012,6 +1019,160 @@ <h3>
           conventions or limitations of the host operating system.
         </p>
       </section>
+      <section>
+        <h3>
+          `file_handlers` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest">`file_handlers`</dfn></code> member is a [=list=] that
+          provides instructions for how the app handles file-opening actions
+          that originate outside of the app itself.
+        </p>
+        <p>
+          The management, presentation and selection of registered
+          file-handling applications is at the discretion of the operating
+          system.
+        </p>
+        <p>
+          To <dfn>process the `file_handlers` member</dfn>, given [=ordered
+          map=] |json:ordered map|, [=ordered map=] |manifest:ordered map|,
+          |manifest_url::URL|:
+        </p>
+        <ol class="algorithm">
+          <li>Let |processedFileHandlers:list| be a new [=list=].
+          </li>
+          <li>Set |manifest|["file_handlers"] to |processedFileHandlers|.
+          </li>
+          <li>If |json|["file_handlers"] doesn't [=map/exist=] or
+          |json|["file_handlers"] is not a [=list=], return.
+          </li>
+          <li>[=list/For each=] |entry:ordered map| of |json|["file_handlers"]:
+            <ol>
+              <li>Let |file_handler:ordered map| be [=process a file handler
+              item=] with |entry|, |manifest/start_url|, |manifest/scope|, and
+              |manifest_url|.
+              </li>
+              <li>If |file_handler| is failure, continue.
+              </li>
+              <li>[=list/Append=] |file_handler| to |processedFileHandlers|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+        <p>
+          On [=installation=], a user agent SHOULD run the process to
+          [=register file handlers=]
+        </p>
+        <section>
+          <h3>
+            Registering file handlers
```

Comment:
Thanks for the clarification, @evanstade. This may become contentious as it may be that there are certain file types that a web application might never be allowed to handle (e.g., system files)... but this may be a UA-level restriction or os-level.

So, question is: would we allow registering anything? or would it be limited to a few well-known types (which yes, massively restricts the utility of this).

Reply:
I don't think there's any _file type_ that we wouldn't want to allow access to. Access control should be applied to individual files that a user shouldn't be able to edit, as you say, by the OS. If a user is allowed to edit a file, I think there's a business justification for wanting to do it via a PWA. If we block specific file types from being edited, we run the risk of forcing apps/users to other ecosystems which don't have this restriction and are much less secure overall. And the greatest threat to user security/privacy is probably in the divulgence of PII or other secrets, which are more likely to be in .doc or .txt or .xlsx or .pdf than some obscure type --- these are clearly types we need to support.